Sources/StreamChatUI/ChatMessageList/ChatChannelHeaderView.swift

```diff
@@ -0,0 +1,144 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// The view that displays channel information on the message list header
+public typealias ChatChannelHeaderView = _ChatChannelHeaderView<NoExtraData>
+
+/// The view that displays channel information on the message list header
+open class _ChatChannelHeaderView<ExtraData: ExtraDataTypes>:
+    _View,
+    ThemeProvider,
+    _ChatChannelControllerDelegate {
+    /// Controller for observing data changes within the channel.
+    open var channelController: _ChatChannelController<ExtraData>?
+
+    /// The user id of the current logged in user.
+    open var currentUserId: UserId? {
+        channelController?.client.currentUserId
+    }
```

Comment:
nit: I wound't publicly expose current user id on a UI component, tbh. Clients can always get it from a chat client/controller

Reply:
This is stored property, it will always get it from the `client`, maybe we can change this to `public` instead of `open`. But this makes the code cleaner when using the current user id, also makes it easier to test the component by overriding `currentUserId` on tests

Reply:
if we expose `currentUserId` on one component customer may start asking why it's missing in other UI components which we're not willing/planning to do 😄

Reply:
if it's a useful helper/is needed in tests we could make it `internal` but exposing identifiers on UI components feels like the wrong thing to me

Reply:
I don't think it is big deal, to be honest, even the customer can add this property themselves. Also will help them for testing. If the customers have custom logic in the components, they will need to test the logic, and this will be very useful for them. This property is just a helper to get the current user from the `client`. I don't see any potential danger with this approach